nasa-veda-platform/scientific-computing/custom-environments.qmd

```diff
@@ -0,0 +1,171 @@
+the `latest` tag, so a tag must be supplied.
+
+When the hostname of the image registry is not specified, it defaults to
+`docker.io`, the canonical hostname of the official Docker registry.  Therefore,
+the following 2 image identifiers are equivalent:
+
+* `pangeo/pangeo-notebook:latest` (as shown in the screenshot above)
+* `docker.io/pangeo/pangeo-notebook:latest` (including a registry hostname prefix)
+
+If you wish to use an image hosted in a regsistry other than `docker.io`, you
+must explicitly specify the hostname of the registry.  Note the intervening `/`
+between the hostname and the remainder of the identifier shown in the second
+bulleted item above.
+
+Once you have specified your custom image, choose your desired **Resource
+Allocation** and press the **Start** button at the bottom of the form to launch
+your environment.
+
+## Environment: Build your own image
+
+When you wish to use a custom environment, but are not aware of a publicly
+available Docker image that suits your needs, the **Environment** option **Build
+your own image** provides a relatively easy and convenient means for building
+and using your desired environment (image), without requiring knowledge of
+Docker.
+
+This option leverages the
+[repo2docker](https://repo2docker.readthedocs.io/en/stable/index.html) tool to
+handle the grunt work of building a custom Docker image for you, by allowing you
+to **fully describe your desired image via files in a public code repository**.
+
+::: {.callout-note}
+Currently, the VEDA JupyterHub supports only **GitHub** as a repository provider
+for use with repo2docker.
+:::
+
+### Describing Your Image
+
+The beauty of leveraging **repo2docker** is that it will use the contents of a
+public code repository (repo) to build a Docker image, doing all of the heavy
+lifting for you.  You do not need to know how to write a `Dockerfile` because it
+generates one for you.  Hence the name *repo2docker*.
+
+In brief, repo2docker allows you to describe the following within a public
+GitHub repository:
+
+- A **list of dependencies** to install, specified via various files (such as
+  `environment.yml`, `requirements.txt`, or `apt.txt`, among others), and are
+  installed **in addition to** a [base set of dependencies].
+- *Optionally*, a **post-build script** to run after all dependencies are
+  installed (this must be named [postBuild])
+- *Optionally*, a **pre-session script** to run each time you start a server
+  with your image (this must be named [start])
+
+Such files from your repository are used to construct a `Dockerfile` file on
+your behalf, so you don't have to write one yourself, unless your really want
+to.  If your repo contains a `Dockerfile`, repo2docker will ignore all other
+files in your repo that it would otherwise use to generate a `Dockerfile`, and
+simply use the `Dockerfile` from your repo.
+
+For complete details on what repo2docker supports, see the following:
+
+- [Where to put configuration files]
```

Comment:
Wondering if the order of links should be changed

Reply:
Do you have a suggestion? Are you thinking of reversing the first 2? Put the "what" before the "where"?

Reply:
exactly, what before where